filesystem.go

```diff
@@ -58,6 +65,29 @@ func New(fs billy.Filesystem, path string) SivaFS {
 	}
 }
 
+// NewFilesystem creates an entire filesystem using siva as the main backend,
+// but supplying unsupported functionality using as a temporal files backend
+// the main filesystem
```

Comment:
- _Is it ok if path is "/tmp" here?_
Depends of the case, we can pass as a parameter the path to be temporal, but I thought that is not necessary.
- _What if fs already has a temp support?, after all, it is full billy.Filesystem._
The custom temporal implementation is mandatory here, because we can create temporal files outside the temporal directory if we set a dir using TempFile method. So, use the temporal implementation in `fs` is not valid for us.
- _Explain that the temporal files are created in the fs filesystem, instead of in memory, for instance._
Is not correctly explained in the constructor documentation? If the filesystem provided is the memory implementation, temporal files will be stored on memory. We can modify the constructor and send a custom filesystem to store temporal files, that could be on memory one.
- _Are temporal files not stored in the siva medium?_
SivaFS doesn't support temporal files. That's is the reason of this PR